File: J_se_les12/HW/Core/Logic/Update/UpdateBase.java
package Java.J_se_les12.HW.Core.Logic.Update;

import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleInput;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public abstract class UpdateBase {
    Phonebook phonebook;
    IConsoleInput input;

    protected UpdateBase(Phonebook phonebook, IConsoleInput input) {
        this.phonebook = phonebook;
        this.input = input;
    }

    public abstract void update(int index);
}

File: J_se_les13/HW/Core/Logic/Update/UpdateBase.cs
using HW.Core.Phonebook;
using HW.Core.Logic.Interfaces;
namespace HW.Core.Logic.Update
{
    public abstract class UpdateBase
    {
        public PhonebookV1 Phonebook { get; set; }

        public IConsoleInput Input { get; set; }
        protected UpdateBase(PhonebookV1 phonebook, IConsoleInput input)
        {
            Phonebook = phonebook;
            Input = input;
        }
        public abstract void Update(int index);
    }
}